pattoo_shared/installation/packages.py

```diff
@@ -136,11 +101,14 @@ def install(requirements_dir, installation_directory, verbose=False):
     # Check for outdated packages
     if verbose:
         print('Checking for outdated packages')
-    installed_packages = get_installed_packages()
-    for key in installed_packages:
-        if version_check(key, installed_packages) is False:
-            # Reinstall updated version of package
-            install_missing_pip3(key, verbose=verbose)
+    for package in lines:
+        # Get packages with versions from pip_requirements.txt
+        requirement_package = package.split('==', 1)
+        if len(requirement_package) == 2:
+            installed_version = get_package_version(requirement_package[0])
+            # Reinstall package if incorrect version is installed
+            if installed_version != requirement_package[1]:
+                install_missing_pip3(package, verbose=verbose)
```

Comment:
You need to be able to specify the specific version of a package in `install_missing_pip3`. If `pip_requirements.txt` requires an older version of a package because the newer version has a bug, then the older one needs to be installed. This is important. The latest version isn't always the best.

Reply:
It actually processes the version listed in the pip_requirements.txt, since the variable, `package`, in the for loop below has the package name and the version attached, as long as the version is specified in the pip_requirements file.

This is the full code block
```
    for package in lines:
        # Get packages with versions from pip_requirements.txt
        requirement_package = package.split('==', 1)
        if len(requirement_package) == 2:
            installed_version = get_package_version(requirement_package[0])
            # Reinstall package if incorrect version is installed
            if installed_version != requirement_package[1]:
                install_missing_pip3(package, verbose=verbose)
```

So since the package name with its version gets passed into `install_missing_pip3`, the package gets reinstalled to reinstalled if the wrong version had been installed before.

This is the output for the check when an older version of pattoo shared is specified in the pip_requirements file.

```
Checking for outdated packages
Running Command: "python3 -m pip install PattooShared==0.0.92 -U --force-reinstall"

```

Reply:
Do your trim out the spaces and replace `>` and `<` with `=` to make this functional?